### ISSUE TITLE
Howto achieve add-cluster-role-to-user

### ISSUE THREAD

**Issue by ridvan70** [openshift/openshift-restclient-java#428]
Hi,

Not an issue but an inquiry:

How to achive the following command with rest-client-java:

`oc adm policy add-cluster-role-to-user self-provisioner -z <accountname> -n cloudmgmt`

Regards
Ridvan

**Comment by adietish** [comment]
@ridvan70 Hi, we had to drop support for administrative operations. This because of the limited resources that we have at our disposal. On the other hand, if you'd be happy to work on such thing we'd be more than happy to assist and support you.

**Comment by ridvan70** [comment]
Hi,

As of working on such thing. How does the process work?

I would commit my solutions. We can begin with my solution/workaround to the subject.

I've achived this by creating a `RoleBinding ` and using a template:

 ```
ModelNode mn = getClusterRoleBinding(#svc_account_name#, #cluster_role#, #namespace#);
RoleBinding rb = new RoleBinding(mn, oc, new HashMap<String, String[]>());
oc.create(rb);
```

```
    private ModelNode getClusterRoleBinding(
            String svcAccountName,
            String clusterRole,
            String namespace) throws IOException {

        InputStream is = getClass().getResourceAsStream("/com/mycompany/templates/ClusterRoleBinding.json");

        String json = IOUtils.toString(is, StandardCharsets.UTF_8.name());

        // metadata.name = #svc_account_name#-#cluster_role#
        json = json.replaceAll("#svc_account_name#", svcAccountName);
        json = json.replaceAll("#cluster_role#", clusterRole);
        json = json.replaceAll("#namespace#", namespace);

        ModelNode mn = ModelNode.fromJSONString(json);

        return mn;
    }

```
ClusterRoleBinding.json
```
{
   "kind":"ClusterRoleBinding",
   "apiVersion":"rbac.authorization.k8s.io/v1",
   "metadata":{
      "name":"#svc_account_name#-#cluster_role#",
      "creationTimestamp":null
   
   },
   "subjects":[
      {
         "kind":"ServiceAccount",
         "name":"#svc_account_name#",
         "namespace":"#namespace#"
      
      }
   
   ],
   "roleRef":{
      "apiGroup":"rbac.authorization.k8s.io",
      "kind":"ClusterRole",
      "name":"#cluster_role#"
   }
}
```

**Comment by adietish** [comment]
Hi @ridvan70,

What you have looks like a great starting point. To be very honest it's a very long time I worked on this area so I'm refreshing my memories with you here. What I see in a quick digging is that known resource types are registered via the properties file in ```src/main/resources/k8stypes.properties```. The association Class->kubernetes schema type is then used to report the class that's required for a given schema type at ```TypeRegistry.getRegisteredType(schemaType)```. Given that the RoleBinding type already exists in this properties file, things should be fine at this level.
```TypeMetaFactory``` is the factory for resource instances. It queries ```TypeRegistry``` and instantiates the class that's associated with a given schema kind. I assume that in your use case you'd want to send a request, you dont want to unmarshall a response. Creating a  type is done via ```TypeMetaFactory.stubKind()```. An example of this is visible in multiple places in the library. I stumbled on the following for scaling requests:

ScaleCapability.scaleTo():
```
        IScale scale = (IScale) factory.stubKind(ARG_KINDS.get(rc.getKind()), Optional.of(rc.getName()),
                Optional.of(rc.getNamespaceName()));
        scale.setSpecReplicas(replicas);
        return client.execute(factory, IHttpConstants.PUT, rc.getKind(), rc.getNamespaceName(), rc.getName(),
                CAPABILITY, null, scale, Collections.emptyMap());
```

```factory``` is the ```TypeMetaFactory``` that we discussed right above. 

Guess that this should help to get further, does it?
Feel free to ping back if I can be of anly further help.

Cheers
André
